src/controllers/usersController.js

```diff
@@ -0,0 +1,57 @@
+import models from '../database/models';
+import auth from '../helpers/authHelper';
+
+/**
+ * UserController class
+ */
+class UsersController {
+  /**
+   *
+   * @param {Object} req - server request
+   * @param {Object} res - server response
+   * @returns {Object} - custom response
+   */
+  static async signUp(req, res) {
+    const {
+      firstname, lastname, username, email, password
+    } = req.body;
+
+    const foundEmail = await models.User.findOne({ where: { email } });
+    const foundUsername = await models.User.findOne({ where: { username } });
+
+    if (foundEmail) {
```

Comment:
Don't you think abstracting this to a function is a good idea? So instead of ```      return res.status(409).json({
        errors: 'username already taken'
      });``` all the time you can do something like `handeResponse(409, { error: 'user with email already exists'})`